### PR TITLE
release v0.6.0

### PR DESCRIPTION
**Pull request by desimone**
## Help test it
```
 pomerium/pomerium:v0.6.0RC1
v0.6.0RC1: digest: sha256:16c075f776ae99740a661f803d3a83362ea44b18a8ad706444465018d9b04512 size: 1152
```

## Related issues

- Replaces #460 

## **Checklist**:

**Before tagging**

- [x] Bump version
- [x] Update doc images for new ui
- [x] Squash pngs
- [x] Fix [algolia not indexing /configuration](https://github.com/algolia/docsearch-configs/commit/b20aaae5388158c047708be844a7c8ca0463451d)
- [x] Update docs to have a new version in config.js and in netlify
- [x] Review and update examples 
- [x] Review and update dockerfiles 
- [x] Review and update docs
- [x] Review and update changelog
- [x] Review and update upgrading guide
- [x] Deprecate helm/stable https://github.com/helm/charts/pull/20146 @travisgroth 

**After tagging**
- [ ] Review and update helm (merge after tag)  https://github.com/pomerium/pomerium-helm/pull/50
- [ ] Update github release docs
- [ ] Set `latest` branch to track the new release
- [ ] Publish tagged version of documentation
- [ ] Review and update helm-charts (merge after tag) 
- [ ] Change docker examples back to `master` on master.

### CODE REVIEW

**Comment by desimone** [comment]
@travisgroth how big of a pain would it be to update the architecture diagrams? Can come after release, but was just realizing

File: docs/configuration/examples/kubernetes/pomerium-cache.yml [hunk 1]
```diff
@@ -0,0 +1,64 @@
+apiVersion: v1
+kind: Service
+metadata:
+  name: pomerium-cache-service
+spec:
+  clusterIP: None # cache is a headless service!
```
Comment (desimone) [inline docs/configuration/examples/kubernetes/pomerium-cache.yml]: 👀 did i do this right?

Comment (travisgroth) [reply docs/configuration/examples/kubernetes/pomerium-cache.yml]: Yep 💯

**Comment by travisgroth** [comment]
>@travisgroth how big of a pain would it be to update the architecture diagrams? Can come after release, but was just realizing

Shouldn't be too bad.  It's just an extra component in two diagrams.

**Comment by travisgroth** [comment]
I was thinking we'd merge https://github.com/helm/charts/pull/20146 before this release.  If nothing else, it should be in the post release checklist.  Thoughts?

**Comment by desimone** [comment]
> I was thinking we'd merge [helm/charts#20146](https://github.com/helm/charts/pull/20146) before this release. If nothing else, it should be in the post release checklist. Thoughts?

💯

**Pull request merged**